pkg/mic/mic.go

```diff
@@ -1317,3 +1327,94 @@ func (c *Client) checkIfIdentityImmutable(id string) bool {
 	}
 	return false
 }
+
+func (c *Client) generateIdentityAssignmentState() (currentState map[string]map[string]bool, desiredState map[string][]string, isVMSSMap map[string]bool, err error) {
+	assignedIDs, err := c.CRDClient.ListAssignedIDs()
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to list AzureAssignedIdentities, error: %+v", err)
+	}
+
+	isVMSSMap = make(map[string]bool)
+	currentState = make(map[string]map[string]bool)
+	desiredState = make(map[string][]string)
+	for _, assignedID := range *assignedIDs {
+		node, err := c.NodeClient.Get(assignedID.Spec.NodeName)
```

Comment:
Can we use a map in this method to keep the `assignedID.Spec.NodeName`, isvmss metadata instead of getting the node for every single `AzureAssignedIdentity`. Although this is read from the informer cache, we could avoid these multiple GETs in case of VMSS as all the nodes will mostly belong to the same VMSS